datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -374,44 +379,38 @@ fn get_sorted_iter(
             })
         })
         .collect::<Result<Vec<_>>>()?;
-    let indices = lexsort_to_indices(&sort_columns, None)?;
+    let indices = lexsort_to_indices(&sort_columns, fetch)?;
 
-    Ok(SortedIterator::new(indices, row_indices, batch_size))
+    //Only maintain composite indexes
+    let row_indices = indices
+        .values()
+        .into_iter()
+        .map(|i| row_indices[*i as usize])
+        .collect();
+
+    Ok(SortedIterator::new(row_indices, batch_size))
```

Comment:
Some cleanup - we can do this immediately instead of keeping it in `SortedIterator`